uint8 node
uint32[] data
